### PR TITLE
v2.5.0 first pass

### PR DESCRIPTION
**Pull request by datamattsson**

### CODE REVIEW

File: docs/csi_driver/index.md [hunk 1]
```diff
@@ -357,20 +374,37 @@ HPE currently supports up to three minor releases of the HPE CSI Driver for Kube
...
-* The HPE CSI Driver does not support access mode transformations. I.e, it's not possible to clone an "RWO" volume into an "RWX" volume or vice versa. See [this note](../partners/redhat_openshift/index.md#storageprofile_for_openshift_virtualization_source_pvcs) on how to work around this limitation with OpenShift Virtualization.
 
 ## iSCSI CHAP Considerations
 
 If iSCSI CHAP is being used in the environment, consider the following.
 
-### CSI driver 1.3.0 and Above
+### Existing PVs and iSCSI sessions
+
+It's not recommended to retro fit CHAP into an existing environment where `PersistentVolumes` are already provisioned and attached. If necessary, all iSCSI sessions needs to be logged out from and the CSI driver Helm chart needs to be installed with [cluster-wide iSCSI CHAP credentials](using.md#cluster-wide_iscsi_chap_credentials) for iSCSI CHAP to be effective, otherwise existing non-authenticated sessions will be reused.
```
Comment (dileepds) [inline docs/csi_driver/index.md:389]: Confused here on this statement. As I understand, we have implemented backward compatibility. User does not need to make any changes unless he wants to move to new CHAP config. Existing CHAP logged in session will continue to work even after moving to 2.5.0, right ?

File: docs/csi_driver/using.md [hunk 2]
```diff
@@ -68,7 +77,7 @@ kubectl kustomize client/config/crd | kubectl create -f-
 kubectl -n kube-system kustomize deploy/kubernetes/snapshot-controller | kubectl create -f-
 ```
 
-```text fct_label="HPE CSI Driver v2.4.0"
+```text fct_label="v2.4.0"
```
Comment (AnushaY1916) [inline docs/csi_driver/using.md:80]: Why is the HPE CSI Driver text removed? Is it because of insufficient space?

Comment (datamattsson) [reply docs/csi_driver/using.md:80]: This is how it renders now. 
image

Spelling it out for each version breaks formatting.

File: docs/csi_driver/index.md (hunk 1 above)
Comment (datamattsson) [reply docs/csi_driver/index.md:389]: This situation discusses enabling CHAP on a cluster that has existing non-authenticated iSCSI sessions and existing PVs.

Comment (dileepds) [reply docs/csi_driver/index.md:389]: ok got it.

**Review by dileepds — DISMISSED** [review]
LGTM

**Review by AnushaY1916 — APPROVED** [review]

**Review by AnushaY1916 — APPROVED** [review]

**Review by dileepds — DISMISSED** [review]
LGTM

**Review by AnushaY1916 — COMMENTED** [review]
Can we specify that there is a risk of data loss when the file system is repaired?

File: docs/csi_driver/using.md [hunk 3]
```diff
@@ -158,18 +135,24 @@ Common HPE CSI Driver `StorageClass` parameters across CSPs.
...
 | csi.storage.k8s.io/fstype     | Text           | Filesystem to format new volumes with. XFS is preferred, ext3, ext4 and btrfs is supported. Defaults to "ext4" if omitted. |
 | fsOwner                       | userId:groupId | The user id and group id that should own the root directory of the filesystem. |
 | fsMode                        | Octal digits   | 1 to 4 octal digits that represent the file mode to be applied to the root directory of the filesystem. |
 | fsCreateOptions               | Text           | A string to be passed to the mkfs command.  These flags are opaque to CSI and are therefore not validated.  To protect the node, only the following characters are allowed:  ```[a-zA-Z0-9=, \-]```. |
+| fsRepair                      | Boolean        | When set to "true", if a mount fails and filesystem corruption is detected, this parameter will control if an actual repair will be attempted. Default: "false". <br/><br />**Note:** `fsRepair` is unable to detect or remedy corrupted filesystems that are already mounted. **Data loss may occur during during the attempt to repair the filesystem.** |
```
Comment (AnushaY1916) [inline docs/csi_driver/using.md:145]: 'during' word mentioned twice

**Review by dileepds — APPROVED** [review]
LGTM

**Pull request merged**